content/post/gke-redis/index.md

```diff
@@ -0,0 +1,504 @@
+
+Once these are handled correctly and the Redis object is deployed, you will see that the following are created:
+
+```bash
+$ kubectl get all -n demo
+NAME                 READY   STATUS    RESTARTS   AGE
+pod/sample-redis-0   1/1     Running   0          2m11s
+
+NAME                        TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+service/sample-redis        ClusterIP   10.8.1.16    <none>        6379/TCP   2m11s
+service/sample-redis-pods   ClusterIP   None         <none>        6379/TCP   2m11s
+
+NAME                            READY   AGE
+statefulset.apps/sample-redis   1/1     2m12s
+
+NAME                                              TYPE               VERSION   AGE
+appbinding.appcatalog.appscode.com/sample-redis   kubedb.com/redis   6.2.5     2m14s
+
+NAME                            VERSION   STATUS   AGE
+redis.kubedb.com/sample-redis   6.2.5     Ready    2m22s
+```
+Let’s check if the database is ready to use,
+
+```bash
+kubectl get redis -n demo
+NAME           VERSION   STATUS   AGE
+sample-redis   6.2.5     Ready    4m
+```
+> We have successfully deployed Redis in GKE. Now we can exec into the container to use the database.
+
+### Accessing Database Through CLI
+
+To access the database through CLI, we have to get the credentials to access. Let’s export the credentials as environment variable to our current shell :
+
+#### Export the Credentials
+
+KubeDB will create Secret and Service for the database `sample-redis` that we have deployed. Let’s check them by following command,
+
+```bash
+$ kubectl get secret -n demo -l=app.kubernetes.io/instance=sample-redis
+NAME                  TYPE                       DATA   AGE
+sample-redis-auth     kubernetes.io/basic-auth   2      7m49s
+sample-redis-config   Opaque                     1      7m49s
+
+$ kubectl get service -n demo -l=app.kubernetes.io/instance=sample-redis
+NAME                TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+sample-redis        ClusterIP   10.8.1.16    <none>        6379/TCP   7m52s
+sample-redis-pods   ClusterIP   None         <none>        6379/TCP   7m52s
+
+```
+Here, we have to use the service `sample-redis` and secret `sample-redis-auth` to connect with the database, and we are going to use `PASSWORD` to authenticate and insert the sample data.
+At first, let’s export the `PASSWORD` as environment variables to make further commands re-usable.
+
+```bash
+$ export PASSWORD=$(kubectl get secrets -n demo sample-redis-auth -o jsonpath='{.data.\password}' | base64 -d)
+```
+
+#### Insert Sample Data
+
+In this section, we are going to login into our Redis database pod and insert some sample data. 
+
+```bash
+$ kubectl exec -it -n demo sample-redis-0 -- redis-cli -a $PASSWORD
```

Comment:
replace `sample-redis-0` with `sample-redis`. as `sample-redis` is the primary service.

Reply:
solved